### ISSUE TITLE
Link wrapping an image does not get focus styling on GlobalHeader

### ISSUE THREAD

**Issue by johnb-sage** [Sage/carbon#6890]
### Description

An image wrapped in Link (e.g. a logo) is not styled when focused on GlobalHeader.

### Reproduction

https://stackblitz.com/edit/parsium-carbon-starter-hfnajo?file=src%2FApp.tsx

### Steps to reproduce

Tab to the Logo, no focus styling is applied.

Also note the focus styling on the text link, seems that box shadow is being incorrectly applied.

### Carbon version

141.4.4

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by johnb-sage** [comment]
@Parsium I feel GlobalHeader wouldn't comfortably support the double focus ring. Since space is tight in this case, a much thinner focus style might be more appropriate?

**Comment by nineteen88** [comment]
@harpalsingh Hey, would you be able to take a look at this and advise on what the focus border should look like. Thanks!

**Comment by harpalsingh** [comment]
@nineteen88 yep so this is the expectation of the focus on a logo visula, so its an internal version of the double border:

![Screenshot 2024-09-04 at 11 24 13](https://github.com/user-attachments/assets/2af9d2d6-04e4-4bb3-bec1-153e64885ffa)

**Comment by nineteen88** [comment]
FE-6802
